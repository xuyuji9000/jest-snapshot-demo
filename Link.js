import React from 'react'
import createReactClass from 'create-react-class'

const STATUS = {
      HOVERED: 'hovered',
      NORMAL: 'normal',
}

const Link = createReactClass({
    getInitialState: function() {
        return {
          class: STATUS.NORMAL,
        }
    },
    _onMouseEnter: function() {
        this.setState({class: STATUS.HOVERED})
    },
    _onMouseLeave: function() {
        this.setState({class: STATUS.NORMAL})
    },
    render: function(){
        return (
          <a
            className={this.state.class}
            href={this.props.page || '#'}
            onMouseEnter={this._onMouseEnter}
            onMouseLeave={this._onMouseLeave}
          >
            {this.props.children}
          </a>
        )
    }

})

export default Link
