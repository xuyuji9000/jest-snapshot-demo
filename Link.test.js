'use strict'

import React from 'react'
import Link from './Link'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
      const tree = renderer
        .create(<Link page="http://www.instagram.com">Instagram</Link>)
        .toJSON();
      expect(tree).toMatchSnapshot();
})
