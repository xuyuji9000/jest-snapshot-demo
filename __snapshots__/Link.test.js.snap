// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="normal"
  href="http://www.intagram.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Instagram
</a>
`;
